app/controllers/api/v0/github_repos_controller.rb

```diff
@@ -32,9 +32,9 @@ def create_octokit_client
 
       def fetched_repo_params
         params[:github_repo] = JSON.parse(params[:github_repo])
-        fetched_repo = @client.repositories.select do |repo|
+        fetched_repo = @client.repositories.detect do |repo|
```

Comment:
`select.first` always gets to the end of the list, `detect` exits the loop once it has found the first match